monitoring/manager_collector.go

```diff
@@ -0,0 +1,89 @@
+package monitoring
+
+import (
+	"os"
+
+	"github.com/sirupsen/logrus"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metricsclientset "k8s.io/metrics/pkg/client/clientset/versioned"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/longhorn/longhorn-manager/datastore"
+)
+
+type ManagerCollector struct {
+	*baseCollector
+
+	kubeMetricsClient *metricsclientset.Clientset
+	namespace         string
+
+	managerUsageCPUMetric    metricInfo
+	managerUsageMemoryMetric metricInfo
+}
+
+func NewManagerCollector(
+	logger logrus.FieldLogger,
+	nodeID string,
+	ds *datastore.DataStore,
+	kubeMetricsClient *metricsclientset.Clientset,
+	namespace string) *ManagerCollector {
+
+	mc := &ManagerCollector{
+		baseCollector:     newBaseCollector("manager", logger, nodeID, ds),
+		kubeMetricsClient: kubeMetricsClient,
+		namespace:         namespace,
+	}
+
+	mc.managerUsageCPUMetric = metricInfo{
+		Desc: prometheus.NewDesc(
+			prometheus.BuildFQName(longhornName, subsystemManager, "usage_cpu_millicpu"),
```

Comment:
mentioned in LEP how about `cpu_usage_millicpu` and `cpu_request_millicpu` basically index by HW component then specific metric.

Reply:
Thank you! Fixed